### PR TITLE
HADOOP-17507. Add install notes GCC9, CMake 3.19

### PR DESCRIPTION
**Pull request by GauthamBanasandra**
* Added instructions for Ubuntu 14.04 and
  Centos 8 regarding installation of GCC 9.3.0
  and CMake 3.19.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 21s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  23m 24s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 31s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 29s |  |  The patch does not generate ASF License warnings.  |
|  |   |  42m 28s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2663/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2663 |
| Optional Tests | dupname asflicense |
| uname | Linux 89ce270fbbf1 4.15.0-128-generic #131-Ubuntu SMP Wed Dec 9 06:57:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / ad483fd66e8 |
| Max. process+thread count | 572 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2663/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: BUILDING.txt [hunk 1]
```diff
@@ -63,6 +63,20 @@ Installing required packages for clean install of Ubuntu 14.04 LTS Desktop:
...
+  $ sudo apt-get -y install g++-9 gcc-9
+  $ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 --slave /usr/bin/g++ g++ /usr/bin/g++-9
+* CMake 3.19
+  $ mkdir -p /opt/cmake \
+    && curl -L https://cmake.org/files/v3.19/cmake-3.19.0.tar.gz > cmake-3.19.0.tar.gz \
+    && tar -zxvf cmake-3.19.0.tar.gz \
+    && cd cmake-3.19.0 \
+    && ./bootstrap \
+    && make -j$(nproc) \
+    && sudo make install
```
Comment (iwasakims) [inline BUILDING.txt]: We need sudo for creating directory under /opt and the /opt/cmake is not used. I think we don't need to apply Dockerfile practice in this instruction. The Protocol Buffers section below is not a good example. It seemed to be copied from Dockerfile and is assuming root privilege.
same for CentOS 8 part.

Comment (GauthamBanasandra) [reply BUILDING.txt]: Done.

**Review by iwasakims — APPROVED** [review]
+1. Thanks, @GauthamBanasandra.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  5s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  23m  2s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 39s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   |  41m 59s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2663/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2663 |
| Optional Tests | dupname asflicense |
| uname | Linux 96dfcbd4bb09 4.15.0-128-generic #131-Ubuntu SMP Wed Dec 9 06:57:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 9801b78edf2 |
| Max. process+thread count | 584 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2663/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: BUILDING.txt (hunk 1 above)
Comment (GauthamBanasandra) [reply BUILDING.txt]: > The Protocol Buffers section below is not a good example. It seemed to be copied from Dockerfile and is assuming root privilege.

I've fixed this, please review - https://github.com/apache/hadoop/pull/2664.

**Pull request merged**